include/deal.II/base/function_spherical.h

```diff
@@ -0,0 +1,109 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__function_spherical_h
+#define dealii__function_spherical_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/std_cxx11/array.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+
+  /**
+   * A base class scalar-valued function $f=f(r,\theta,\phi)$ defined in
+   * spherical coordinates. The convention for angles is the same as in
+   * GeometricUtilities::Coordinates.
+   *
+   * This is an abstract class which wraps transformation of values, gradients
+   * and hessians from spherical coordinates to Cartesian coordinate system.
+   * Therefore derived classes only need to implement those functions in
+   * spherical coordiantes (see svalue(), sgradient() and
+   * shessian() ).
+   *
+   * @note This function is currently only implemented for dim==3 .
+   *
+   * @author Denis Davydov
+   * @date 2017
+   */
+  template <int dim>
+  class Spherical : public Function<dim>
+  {
+  public:
+    /**
+     * Constructor which should be provided with center @p center.
+     */
+    Spherical(const Point<dim> &center);
+
+    /**
+     * Virtual destructor.
+     */
+    virtual ~Spherical();
+
+    virtual double value (const Point<dim> &point,
+                          const unsigned int component = 0) const;
+
+    virtual Tensor<1,dim> gradient (const Point<dim>   &p,
```

Comment:
i would certainly not add formulas, as for `hessian`'s it's a nightmare. Otherwise the doxygen will be take from the parent class, so i don't think there is a need to add anything here.